platform/engineering/request-for-comment/2024-09-23-evidence-collection-and-polling.md

```diff
@@ -0,0 +1,136 @@
+# RFC: Benefits Evidence and Ancillary Documents Handling and Technical Details
+
+- Date: 2024-09-23
+- Related Issue: 
+
+## Background
+Evidence and ancillary forms filled out on veterans/users behalf, are collected, generated, stored, and sent in many different ways across products/forms.
+
+For a better user experience, and for the sake of less technical complexity, some amount of technical and architectural decisions should be agreed upon and implemented across all products and forms. Ideally, where possible, sharing code to reduce complexity and drift. But at the very least should operate similarly. This RFC aims to describe and gain concensus on the full evidence and ancillary form lifecycle. 
+
+## Definitions
+Evidence: Any document uploaded by a user, that vets-api is responisble for transmitting down- or up- stream. Examples could include a medical document, buddy statement, scanned-in filled-out form, digitally filled-out form, and other various documents. 
+
+Ancillary Document: Any document that has an official VA designated form number, that we fill-out and transmit somewhere, based on information the end-user enters into the frontend. Depending on the form/situation, the end-user may or may not know they filled this form out, or that one was filled out on their behalf. Examples include form 21-4142, form 21-0781, and other forms that we take information and stamp onto a PDF to transmit.
+
+Documents: May be used to refer to both evidence and ancillary documents collectivly.
+
+
+## Motivation
+
+* Reduced code complexity
+* Align on shared strategy
+* Consistent fronend experience for the end users
+* Elimination of silent failure (through status polling)
+
+
+## Design
+
+### Frontend Accepting of Benefits documents
+#### Validation
+Validation should be kept to a minimum on the frontend only to the most obvious issues, such as:
+* Filesize not exceeding X
+* Document extension supported
+* Document not password protected (or password collected if so, since we NEED to check this here)
+
+All other checks should be done by the system downstream that is ultimately responsible for accepting the file or not. We should not try to maintain separate lists of requirements that can get out of sync. We should, after initial validation above, check the file against a validation endpoint for the downstream system. 
+
+You may think, "why not just actually upload it here and rely on the error checking there, instead of uploading twice (once to validate and once to upload)". In places where this is possible, this should be done, however many evidence documents cannot be uploaded without first knowing a claim ID which must be sent with the document to be uploaded, which is not yet known at document upload time, and only later known after initial form submission. 
```

Comment:
> many evidence documents cannot be uploaded without first knowing a claim ID which must be sent with the document to be uploaded

Is this also true with the Claim Evidence API? My naïve read of their Swagger doc makes it seem like [file upload](https://claimevidence-api-test.dev.bip.va.gov/api/v1/rest/swagger-ui.html#/File/upload) and [associating a file with a claim ID](https://claimevidence-api-test.dev.bip.va.gov/api/v1/rest/swagger-ui.html#/Association/associate) are separate endpoints.

Reply:
Re-adding some context from Slack - in the near term we're using LH & polling, but in the longer term, the enablement team is interested in whether directly using BIP Claims Evidence API would be an improvement for our products with regards to developer experience, product simplicity, more features, less middleware, etc.

Reply:
Im not familiar with the BIPS api very much, Ive never used it, but the Lighthouse endpoints that do use it, uploads are comprised of 2 calls, one to plop in the file to the eFolder, and another to associate said file with a particular claim ID. 

For 526 related docs, https://developer.va.gov/explore/api/benefits-documents/docs?version=current Benefits documents api, requires a claim ID.  One portion of what the call does, which is upload to efolder, might not require that, but the call overall that va.gov makes to lighthouse, often requires it. 

Could we bypass lighthouse and make our own calls directly to BIPS or other apis? Sure. But we've been given direction to prefer/standardize to using lighthouse when possible/supported, from what I am aware.